docs/DafnyRef/Grammar.md

```diff
@@ -89,10 +89,9 @@ let you reconstruct the original grammar.
 
 ## 2.1. Dafny Input {#sec-unicode}
 
-Dafny source code files are readable text encoded as UTF-8 Unicode
-(because this is what the Coco/R-generated scanner and parser read).
+Dafny source code files are readable text encoded in UTF-8, where each decoded character has to be from the Basic Multilingual Plane and therefore encodable with a single UTF-16 code unit. This is what the Coco/R-generated scanner and parser read.
 All program text other than the contents of comments, character, string and verbatim string literals
-consists of printable and white-space ASCII characters,
+consists of printable and white-space ASCII characters.
```

Comment:
The change from , to . is wrong here -- it leaves a sentence fragment on the next line.

Reply:
Can you fix it in your next PR please? I can review that. Thanks for reporting.

Reply:
Added to cok-docs-TODOs, PR#2817